src/main/steps/applicant1/application-ended/get.ts

```diff
@@ -0,0 +1,44 @@
+import autobind from 'autobind-decorator';
+import { Response } from 'express';
+
+import { getSystemUser } from '../../../app/auth/user/oidc';
+import { getCaseApi } from '../../../app/case/CaseApi';
+import { getUnreachableAnswersAsNull } from '../../../app/case/answers/possibleAnswers';
+import { ApplicationType, SWITCH_TO_SOLE } from '../../../app/case/definition';
+import { AppRequest } from '../../../app/controller/AppRequest';
+import { GetController } from '../../../app/controller/GetController';
+
+import { generateContent } from './content';
+
+@autobind
+export default class ApplicationEndedGetController extends GetController {
+  constructor() {
+    super(__dirname + '/template', generateContent);
+  }
+
+  public async get(req: AppRequest, res: Response): Promise<void> {
+    const caseworkerUser = await getSystemUser();
+    req.locals.api = getCaseApi(caseworkerUser, req.locals.logger);
+
+    try {
+      req.session.userCase.applicationType = ApplicationType.SOLE_APPLICATION;
```

Comment:
i feel like this whole logic should be in the post controller for the previous page.
user clicks switch to sole -> trigger event in post -> redirect to application ended.

Reply:
The flow that leads the user to this page is the log in page that is handled by Idam. The user logs in after app2 has said No the marriage hasn't broken down. Therefore there is no post on a page prior to this - only the log in page. If there was was a page before then, yes I would have done that.

Reply:
fair